lib/auth/oauth/oauth.ts

```diff
@@ -116,9 +136,17 @@ export function callback(config: ConfigInterface) {
     ).searchParams;
     const shop = query.get('shop')!;
 
+    const response = {} as NormalizedResponse;
+    if (isbot(request.headers['User-Agent'])) {
+      logForBot({request, log, func: 'callback'});
+      throw new ShopifyErrors.BotActivityDetected(
+        'Invalid OAuth callback initiated by bot',
+      );
+    }
```

Comment:
@teddyhwang New callback bot handling.
Your last comment was on an out dated diff. Lots of back and forth on this with @mkevinosullivan and @paulomarg to decide the best course of action.
This is where we've landed.

Next step is to update the express app middleware to handle this error.